classes/SnippetLoader.php

```diff
@@ -0,0 +1,168 @@
+<?php namespace Winter\Pages\Classes;
+
+use ApplicationException;
+use Cache;
+use Carbon\Carbon;
+use Cms\Classes\CmsException;
+use Cms\Classes\Controller as CmsController;
+use Cms\Classes\Page as CmsPage;
+use Cms\Classes\Theme;
+use Cms\Classes\ComponentManager;
+use Exception;
+use Flash;
+use Session;
+use SystemException;
+use Winter\Pages\Classes\SnippetManager;
+
+class SnippetLoader
+{
+    protected static $pageSnippetsCache = [];
+
+    /**
+     * Add a component registered as a snippet to the active controller.
+     *
+     * @throws SystemException
+     * @throws CmsException
+     */
+    public static function registerComponentSnippet(array $snippetInfo): string
+    {
+        $controller = CmsController::getController();
+
+        // Make an unique alias for this snippet based on its name and parameters
+        #$snippetInfo['code'] = uniqid($snippetInfo['code'] . '_' . md5(serialize($snippetInfo['properties'])) . '_');
+        // the line above was commented out to allow the overriden partials in theme to be used for the component alias
```

Comment:
@mjauvin could you explain this further to me?

Reply:
It's been a while and I don't recall specifically, but the code provided in the content (richeditor or elsewhere) cannot be modified, otherwise it breaks.

Reply:
Ok, I experimented a bit and the problem is if you modify the snippet code as above, the overridden markup is not found in the theme because the "alias" is totally different.

For example, let's say you have a component named "FAQ" and it's used as a snippet, if the theme creates a "partials/faq/default.htm" file to override the default component markup, obfuscating the snippet code with `uniqid($snippetInfo['code'] . '_' . md5(serialize($snippetInfo['properties'])) . '_');` will prevent that overriden markup to be found (it would search "partials/{uniqueid}" which obviously won't exist.

Reply:
And I'm guessing the original reason for this code was to support multiple instances of the same component provided snippet in the same field?

Reply:
Using different snippet code would achieve the same goal.